### ISSUE TITLE
How to debug code inside the RPC server

### ISSUE THREAD

**Issue by oren0e** [0rpc/zerorpc-python#270]
Hi, 
I would like a way to "attach" to the rpc server and be able to step through the code that is being run by it in a debugger. How can I do that? Currently the code that runs in the RPC server is kind of a black box to me and this makes it hard to debug problems.

Thanks in advance.

**Comment by oren0e** [comment]
@bombela I know, I have tried to debug with pdb (the normal debugger I usually use) but the problem is that eventually you get to a `.call()` method or something like that for the rpc server and that's it.

**Comment by bombela** [comment]
I might have miss understood. At the risk of stating the obvious, you must
run the debug on the server. Likely with a breakpoint of the handler of
interest.

**Comment by oren0e** [comment]
@bombela of course. The setup is that on the same server there is a code that executes and then calls to the rpc server which is on the same server, just a different process on a different port. My question is how to attach to that port and be able to debug properly.

**Comment by bombela** [comment]
I meant to run the debugger on the python process receiving calls from the
network, and executing the functions/handlers.

If I understood correctly, in your case you would want to run PDB on the
RPC server.

I don't think PDB can attach. But you can surely start the process under
PDB at least. https://docs.python.org/3/library/pdb.html
